hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/ssl/ReloadingX509TrustManager.java

```diff
@@ -138,7 +139,8 @@ X509TrustManager loadTrustManager(CertificateClient caClient)
     X509TrustManager trustManager = null;
     KeyStore ks = KeyStore.getInstance(type);
     ks.load(null, null);
-    ks.setCertificateEntry(rootCACertId, rootCACert);
+    insertCertsToKeystore(caClient.getAllRootCaCerts(), ks);
+    insertCertsToKeystore(caClient.getAllCaCerts(), ks);
```

Comment:
With the certificate bundle supported,  we only need root CA certificates in the trust store, right?

Reply:
For regular cert clients yes, for SCM client no. In the old functionality the SCM client returns null for getRootCaCertificate, and after that uses the sub ca. HA-secure acceptance test immediately fail if we don't have the CA certs here. I believe the functionality where SCM returns null for getting the root CA should be changed. This could be addressed in a separate issue.

Reply:
I know that SCM is different.  What I suggest is try something like 
insertCertsToKeystore(caClient.getAllRootCaCerts() != null ? caClient.getAllRootCaCerts() : caClient.getAllCaCerts(), ks);

Otherwise, except SCM, all other service will add both Root CA and Sub-CA CA certificates in the trust store.